### ISSUE TITLE
Discord Rich Presence Ubuntu is not working

### ISSUE THREAD

**Issue by JackRhodes** [runelite/runelite#5862]
**Describe the bug**
Discord Rich Prescience is not showing RuneLite when using Ubuntu 18.04

**To Reproduce**
1. Run RuneLite using 'java -jar runelitefile.jar'
2. Open Discord

**Expected behavior**
Discord rich presence should display RuneLite as running and my activities in the game. Currently it does not show anything.

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04
 - RuneLite version: 1.4.20
 - Launcher version: 1.6.0

**Comment by Veeeeeeeee** [comment]
From my understanding, I experience the same issue however it's only because I open RuneLite then open Discord after, thus Rich Presence isn't showing I'm playing RuneLite. Just open a second client and it should trigger the Rich Presence, if not I apologise but hopefully it helps your situation.

**Comment by Karolcz125** [comment]
![image](https://user-images.githubusercontent.com/15970653/46581235-7c29ed00-ca35-11e8-9e3f-81aa76784cb1.png)

Discord does not discover any programs on my linux (arch), check if you have got the same problem on Ubuntu because it may be the source of the issue

**Comment by deathbeam** [comment]
It is definitely working, otherwise duplicate of #5719

**Comment by Hakej** [comment]
How is that a duplicate of #5719 if the mentioned issue is on Windows and the problem is on Linux?
I'm having the same issue right now.
Discord, as well as Runelite, are both running with no root privileges.
I'm using Manjaro with Gnome DE.

**Comment by kblicharski** [comment]
I'm having the same issue on Ubuntu 18.10.

**Comment by Nightfirecat** [comment]
How do you have both applications installed? (native app? snap? other?)

**Comment by MrCarroll** [comment]
Both Ubuntu and Manjaro highlight the Snap in their package managers, so it's quite possible they're using it. In the case of Snap, the sandbox model in use prevents RuneLite communicating with the Discord Unix socket, I can look into if there's an easy remedy; but the current scenario is it just straight up won't work by design and actually fixing the problem might not be easily done without Discord involvement or Snapd changes outside the scope of RuneLite itself.

Either way, the Snap problem is outside the scope of this specific issue given the Snap package didn't exist when this was made, although OP 2 years ago may have used the Snap Discord package, which would have the inverse of this problem.

Tl;Dr rich presence + Snap is a tough problem whilst Discord continues to avoid using Dbus

**Comment by kblicharski** [comment]
I have RuneLite installed through the AppImage and Discord through the Snap. Is the solution to the problem then to install Discord using the .deb package? Ubuntu 18.10 has been dead for a while and I can't install one of the dependencies that Discord needs through apt, or else I would test that myself. I guess I'll have to upgrade soon then.

**Comment by Nightfirecat** [comment]
Yes, the solution is to install both as native apps, not running through snap. As James's comment mentions, snap does not allow dbus access, which is what discord uses to report rich status.
